src/dns_resolver_ares.c

```diff
@@ -126,36 +140,60 @@ static void query_completed_cb(void *arg, int status, int timeouts, struct hoste
             dns->is_complete = true;
             dns->in_progress = false;
         }
-        else
+#ifdef IPV6_ENABLED
+        else if (he->h_addrtype == AF_INET6)
         {
             ptr = dns->addrInfo;
-            
-            ptr->ai_addr = calloc(1, sizeof(struct sockaddr_in));
+
+            ptr->ai_addr = calloc(1, sizeof(struct sockaddr_in6));
+            addr6 = (void *)ptr->ai_addr;
```

Comment:
Why not casting to `struct sockaddr_in6*` ?

Reply:
I see, you just followed what was done for IPv4

Reply:
I'm not very happy with all the casting either. A `freeaddrinfo` call is recommended to deallocated these records.
See https://gist.github.com/CIPop/e29729d1c8d6871372f38c8cbcc43f93#file-client_ipv4_ipv6-c-L75